resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -62,49 +137,300 @@
-                "kms:CreateGrant",
-                "s3:GetBucketTagging",
+                "ec2:DescribeVpcPeeringConnections"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VpcAttribute",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeVpcAttribute"
+            ],
+            "Resource": "arn:aws:ec2:*:*:vpc/*"
+        },
+        {
+            "Sid": "VpcEndpointService",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeVpcEndpointServicePermissions"
+            ],
+            "Resource": "arn:aws:ec2:*:*:vpc-endpoint-service/*"
+        },
+        {
+            "Sid": "S3object",
+            "Effect": "Allow",
+            "Action": [
                 "s3:GetObjectAcl",
-                "s3:GetObjectTagging",
-                "s3:ListAllMyBuckets",
+                "s3:GetObjectTagging"
+            ],
+            "Resource": "arn:aws:s3:::*"
+        },
+        {
+            "Sid": "Tiros",
+            "Effect": "Allow",
+            "Action": [
                 "tiros:CreateQuery",
                 "tiros:GetQueryAnswer",
-                "tiros:GetQueryExplanation",
+                "tiros:GetQueryExplanation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "NetworkVerifier1",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:CreateTags"
+            ],
+            "Resource": "arn:aws:ec2:*:*:instance/*",
+            "Condition":{
+                "StringEquals":{
+                    "ec2:CreateAction":"RunInstances",
+                    "aws:RequestTag/red-hat-managed": "true"
+                },
+                "ForAllValues:StringEquals": {
+                    "aws:TagKeys": "red-hat-managed"
+                }
+            }
+        },
+        {
+            "Sid": "NetworkVerifier2",
+            "Effect": "Allow",
+            "Action": [
                 "ec2:RunInstances",
                 "ec2:DescribeInstanceTypes",
-                "ec2:CreateTags",
                 "ec2:GetConsoleOutput"
             ],
+            "Resource": "arn:aws:ec2:*:*:instance/*",
+            "Condition": {
+                "StringEquals": {
+                    "aws:ResourceTag/red-hat-managed": "true"
+                }
+            }
+        },
+        {
+            "Sid": "Ip",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeCoipPools",
+                "ec2:DescribeIpv6Pools",
+                "ec2:DescribePublicIpv4Pools",
+                "ec2:GetAssociatedIpv6PoolCidrs"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "InternetGateway",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeEgressOnlyInternetGateways",
+                "ec2:DescribeInternetGateways"
+            ],
+            "Resource": "*"
```

Comment:
Any way these can be restricted more?

Reply:
doesn't seem like it could be more restricted (checked with https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonec2.html#amazonec2-policy-keys) and the BYOVPC does not have any specific tags